odl/operator/operator.py

```diff
@@ -206,29 +395,64 @@ class Operator(with_metaclass(_OperatorMeta, object)):
     implementation of the respective other is provided.
     """
 
+    def __new__(cls, *args, **kwargs):
+        """Create a new instance."""
+        instance = super().__new__(cls)
+
+        call_has_out, call_out_optional, _ = _dispatch_call_args(cls)
+        instance._call_has_out = call_has_out
+        instance._call_out_optional = call_out_optional
+        if not call_has_out:
+            # Out-of-place _call
+            instance._call_in_place = partial(_default_call_in_place,
+                                              instance)
+            instance._call_out_of_place = instance._call
+        elif call_out_optional:
+            # Dual-use _call
+            instance._call_in_place = instance._call
+            instance._call_out_of_place = instance._call
+        else:
+            # In-place only _call
+            instance._call_in_place = instance._call
+            instance._call_out_of_place = partial(_default_call_out_of_place,
+                                                  instance)
+
+        return instance
```

Comment:
One thought I just got when I commented on the remark above: if we inject the changed/wrapped caller into the *class* before creation, then every subclass (I should check if this is true, but I think so) will inherit this wrapped caller *even if it overrides `__new__`*, while for the current pattern, this won't happen.
Not sure what this means in practice.